test/com/google/enterprise/adaptor/DocumentHandlerTest.java

```diff
@@ -2481,6 +2487,545 @@ public void testPercentEncoding() {
                  + "%2A%28%29%5B%5D%7B%7D%C3%AB%01", encoded);
   }
 
+  // tests for hostsSkipCertCheck
+
+  // server.fullAccessHosts=NOT_OUR_IP_ADDRESS
+  // server.skipCertCheckHosts=NOT_OUR_IP_ADDRESS
+  // adaptor trusts to CN=localhost
+  // requestor ip=127.0.0.3
+  // requestor CN=localhostbad
+  // result: 403, no headers
+  @Test
+  public void testSkipCertCheckBadCNBadIP() throws Exception {
```

Comment:
I think we can come up with more precise word than "bad". For example "untrusted" fits here. Other words may fit in other circumstances.

How about calling this method:
testHttpsConnectionFromUntrustedCN

Why would the IP matter at all?